File: VzBaseLibrary/src/main/java/com/eleven/baselibrary/activity/VzBaseActivity.java
package com.eleven.baselibrary.activity;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.eleven.baselibrary.utils.ActivityManager;

import java.lang.ref.WeakReference;

/**
 * Created by huangwz on 2017/6/3.
 *
 * @Link: vaitt_joy@163.com
 *
 */

public abstract class VzBaseActivity extends AppCompatActivity implements IBaseActivity{

    /**
     * 当前Activity的弱引用，防止内存泄露
     **/
    protected WeakReference<Activity> mContext = null;
    /**
     * 日志输出标志
     **/
    protected final String TAG = this.getClass().getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mContext = new WeakReference<Activity>(this);
        ActivityManager.pushTask(mContext);
        initView(savedInstanceState);
        initData();
        initEvent();
    }

    /**
     * 初始数据
     */
    protected void initData(){}

    /**
     * 初始化控件的事件
     */
    protected void initEvent(){}

    @Override
    protected void onResume() {
        super.onResume();
        resume();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        destroy();
        ActivityManager.removeTask(mContext);
    }
}

File: app/src/main/java/com/vz/phpkotlin/ui/base/VzBaseActivity.kt
package com.vz.phpkotlin.ui.base

import android.app.Activity
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.vz.phpkotlin.utils.ActivityManager
import java.lang.ref.WeakReference

/**
 * Created by huangwz on 2017/4/25.
 */

abstract class VzBaseActivity : AppCompatActivity(), IBaseActivity {

    protected var context: WeakReference<Activity>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        context = WeakReference<Activity>(this)
        ActivityManager.pushTask(context!!)
        initView()
        initData()
        initEvent()
    }

    protected fun initData() {}

    protected fun initEvent() {

    }

    override fun onResume() {
        super.onResume()
        resume()
    }

    override fun onDestroy() {
        super.onDestroy()
        destroy()
    }
}
